docs/batch-system-integration.md

```diff
@@ -344,6 +344,77 @@ JOB_ROUTER_ENTRIES @=jre
 @jre
 ```
 
+### Setting job environments ###
+
+HTCondor-CE offers two different methods for setting environment variables of routed jobs:
+
+- `CONDORCE_PILOT_JOB_ENV` configuration, which should be used for setting environment variables for all routed jobs to
+  static strings.
+- `set_default_pilot_job_env` job route configuration, which should be used for setting environment variables:
+    - Per job route
+    - To values based on incoming job attributes
+    - Using [ClassAd functions](https://htcondor.readthedocs.io/en/latest/misc-concepts/classad-mechanism.html#predefined-functions)
+
+Both of these methods use the new HTCondor format of the
+[environment command]((https://htcondor.readthedocs.io/en/latest/man-pages/condor_submit.html#index-14)), which is
+described by environment variable/value pairs separated by whitespace and enclosed in double-quotes.
+For example, the following HTCondor-CE configuration would result in the following environment for all routed jobs:
+
+``` tab="Configuration"
+CONDORCE_PILOT_JOB_ENV = "WN_SCRATCH_DIR=/nobackup/ http_proxy=proxy.wisc.edu"
+```
+
+```bash tab="Result"
+WN_SCRATCH_DIR=/nobackup/
+http_proxy=proxy.wisc.edu
+```
+
+Contents of `CONDORCE_PILOT_JOB_ENV` can reference other HTCondor-CE configuration using HTCondor's configuration
+[$() macro expansion](https://htcondor.readthedocs.io/en/stable/admin-manual/introduction-to-configuration.html#configuration-file-macros).
+For example, the following HTCondor-CE configuration would result in the following environment for all routed jobs:
+
+``` tab="Configuration"
+LOCAL_PROXY = proxy.wisc.edu
+CONDORCE_PILOT_JOB_ENV = "WN_SCRATCH_DIR=/nobackup/ http_proxy=$(LOCAL_PROXY)"
+```
+
+```bash tab="Result"
+WN_SCRATCH_DIR=/nobackup/
+http_proxy=proxy.wisc.edu
+```
+
+To set environment variables per job route, based on incoming job attributes, or using ClassAd functions, add
+`set_default_pilot_job_env` to your job route configuration.
+For example, the following HTCondor-CE configuration would result in this environment for a job with these attributes:
+
+``` tab="Configuration" hl_lines="5 6 7" 
+JOB_ROUTER_ENTRIES @=jre
+[
+  TargetUniverse = 5;
+  name = "Local_Condor";
+  set_default_pilot_job_env = strcat("WN_SCRATCH_DIR=/nobackup",
+                                     " PILOT_COLLECTOR=", JOB_COLLECTOR,
+                                     " ACCOUNTING_GROUP=", toLower(JOB_VO));
+]
+@jre
+```
+
+``` tab="Job"
+JOB_COLLECTOR = "collector.wisc.edu"
+JOB_VO = "GLOW"
+```
+
+``` bash tab="Result"
+WN_SCRATCH_DIR=/nobackup/
+PILOT_COLLECTOR=collector.wisc.edu
+ACCOUNTING_GROUP=glow
+```
+
+!!!tip "Debugging job route environment expressions"
+    While constructing `set_default_pilot_job_env` expressions, try wrapping your expression in
+    [debug()](#debugging-routes) to help with any issues that may arise.
```

Comment:
Where does debug() output show up?

Reply:
In the JobRouterLog as mentioned in the linked section: https://htcondor-ce.readthedocs.io/en/latest/batch-system-integration/#debugging-routes